sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -59,57 +54,75 @@ class EthService(val stateView: AccountStateView, val nodeView: CurrentView[Acco
     if (blockId == null) return null
     val block = nodeView.history.getBlockById(blockId)
     if (block.isEmpty) return null
-    val transactions = getTransactionsFromBlock(block.get())
-    val txList: ListBuffer[Any] = ListBuffer()
-    transactions.foreach(transaction => {
-      if (hydratedTx) {
-        val receipt = stateView.getTransactionReceipt(Numeric.hexStringToByteArray(Numeric.cleanHexPrefix(transaction.id)))
-        if (!receipt.isEmpty) txList.append(new EthereumTransactionView(receipt.get, transaction))
-      } else txList.append(Numeric.prependHexPrefix(transaction.id))
-    })
-    new EthereumBlock(Numeric.prependHexPrefix(Integer.toHexString(nodeView.history.getBlockHeight(blockId).get())), Numeric.prependHexPrefix(blockId), txList.toList.asJava, block.get())
-  }
+    // TODO: why does this not compile?
+    //val transactions = block.get().transactions.collect { case tx: EthereumTransaction => tx }
```

Comment:
I don't understand why this does not compile - shouldn't it be equivalent to the code below..?

Reply:
Scala patterning matching is not available here, because the AccountTx and EthTx are not covariant.
It is expected.